src/Registration/cReg/NiftiImageData.cpp

```diff
@@ -220,7 +220,7 @@ void NiftiImageData<dataType>::construct_NiftiImageData_from_complex_im_real_com
 
     auto &it_in = in_sptr->begin();
     auto &it_out = out_sptr->begin();
-    for (; it_in!=in_sptr->end(); ++it_in, ++it_out)
+    for (; it_out != out_sptr->end(); ++it_in, ++it_out)
```

Comment:
not sure why this is. Aren't they the same size?

Reply:
no, the input image dimensions are (2, 256, 256) and the output (1, 256, 256) because the first dimension of a `NiftiImageData` object is z-dimension, which is 1 for both input and output images, but the input image is reconstructed from MR acquisition data with two repetitions.

Reply:
Are we (silently) constructing a single volume from a repeated volume? That sounds wrong. I would suggest that we have a chechk that dimensions are identical (e.g. via `get_geometrical_info()` ) and if not throw an error.

Where is the call then that you're trying to fix?

Reply:
Throwing an error to avoid segfault is not very helpful, I believe - I would rather issue a warning that the complex data contains more than one image, and only the first one is going to be used. @paskino which way would you prefer?

> Where is the call then that you're trying to fix?

In `Reg.NiftiImageData.construct_from_complex_image`, if I understand your question correctly

Reply:
I'm trying to understand if this is a "conceptual" problem (i.e. what should we do with weird data) or a bug (i.e. we know how to handle the data, but the code doesn't do it).

What data do we stick into `Reg.NiftiImageData.construct_from_complex_image` where it throws this error? Is it a normal complex image, and If it is complex data with 2 repetitions, then I would argue that the output should be a real image with 2 repetitions, not silently throw away the 2nd repetition.

Unfortunately, this might be getting this into our problem of encoding MRI dimensions. From your description, I understand that data of a single slice with 2 repetitions is encoded as a 3D NiftiImageData. That in itself would be wrong (it should be a 4D image, with a single slice). And as I stated above, the `construct_from` functions really should check that all args have the same geometry. If they don't, an error should be thrown as it is invalid usage. Otherwise other crashes are unavoidable. 

I might misunderstand this completely of course...